arangod/Cluster/ClusterTrxMethods.cpp

```diff
@@ -297,7 +297,11 @@ Future<Result> commitAbortTransaction(arangodb::TransactionState* state,
                       << "synchronous replication: could not drop follower " << follower
                       << " for shard " << cc->vocbase().name() << "/" << tc.collectionName()
                       << ": " << r.errorMessage();
-                  res.reset(TRI_ERROR_CLUSTER_COULD_NOT_DROP_FOLLOWER);
+                  if (res.is(TRI_ERROR_CLUSTER_NOT_LEADER)) {
+                    THROW_ARANGO_EXCEPTION(TRI_ERROR_CLUSTER_SHARD_LEADER_RESIGNED);
+                  } else {
+                    THROW_ARANGO_EXCEPTION(TRI_ERROR_CLUSTER_COULD_NOT_DROP_FOLLOWER);
+                  }
```

Comment:
Is it intentional to throw an exception here?
This code is in a loop that iterates over all collections in the transaction.
Now we may abort after dropping the first follower we could not remove, and don't even try removing others even though that may be necessary.

Reply:
Yes, as it rolls back the local transaction. This behaviour has been in place since before this fix.

Reply:
Is it intentional to drop out of the loop over all followers when removing one of the followers fails? Shall the loop continue and try to remove the other followers too, and only fail afterwards?

Reply:
Plus, I think it _is_ a behavior change. Because previously the loop continued, and now it is aborted early.